packages/harmony/src/components/select/Select/Select.tsx

```diff
@@ -1,253 +1,210 @@
+  props: SelectProps<Value>,
+  ref: Ref<HTMLInputElement>
 ) {
   const {
-    selection: selectionProp,
+    value: valueProp,
     options,
-    optionsLabel,
-    popupAnchorOrigin = { horizontal: 'left', vertical: 'bottom' },
-    popupMaxHeight,
-    popupTransformOrigin = { horizontal: 'left', vertical: 'top' },
-    popupPortalLocation,
-    popupZIndex,
-    ...selectInputProps
+    menuProps,
+    onChange,
+    onClick,
+    clearable,
+    children,
+    ...other
   } = props
 
-  const [selection, setSelection] = useControlled({
-    controlledProp: selectionProp,
+  const { label } = other
+
+  const [value, setValue] = useControlled({
+    controlledProp: valueProp,
     defaultValue: null,
     stateName: 'selection',
     componentName: 'Select'
   })
 
-  // TODO: implement filtering
-  // The state management is already done because this was copied from FilterButton
-  // but I would like to support filtering the options by typing in the SelectInput
-  const [filterInputValue, setFilterInputValue] = useState('')
-  const selectedOption = options.find((option) => option.value === selection)
+  const [isOpen, setIsOpen] = useState(false)
+  const [inputValue, setInputValue] = useState('')
+  const selectedOption = options.find((option) => option.value === value)
   const selectedLabel = selectedOption?.label ?? selectedOption?.value
+  const anchorRef = useRef<HTMLDivElement>(null)
   const inputRef = useRef<HTMLInputElement>(null)
-  const optionRefs = useRef([])
+  const optionRefs = useRef<HTMLButtonElement[]>([])
   const scrollRef = useRef<HTMLDivElement>(null)
 
-  const handleOptionSelect = useCallback(
-    (
-        handleChange: (value: string, label: string) => void,
-        setIsOpen: (isOpen: boolean) => void
-      ) =>
-      (option: SelectOption) => {
-        setSelection(option.value)
-        handleChange(option.value, option.label ?? '')
-        setIsOpen(false)
-      },
-    [setSelection]
+  const handleChange = useCallback(
+    (value: Value) => {
+      setValue(value)
+      setInputValue('')
+      onChange?.(value)
+      setIsOpen(false)
+    },
+    [onChange, setValue]
   )
 
-  const handleOpen = useCallback(() => {
-    // Focus the input after the popup is open
-    setTimeout(() => {
-      inputRef.current?.focus({ preventScroll: true })
-    }, 0)
-  }, [inputRef])
+  useEffect(() => {
+    if (isOpen) {
+      // Focus the input after the popup is open
+      setTimeout(() => {
+        inputRef.current?.focus({ preventScroll: true })
+      }, 0)
+    }
+  }, [isOpen])
 
   const filteredOptions = useMemo(
     () =>
-      options.filter(({ label }) => {
+      options.filter(({ value, label }) => {
         return (
-          !filterInputValue ||
-          label?.toLowerCase().includes(filterInputValue.toLowerCase())
+          !inputValue ||
+          label?.toLowerCase().includes(inputValue.toLowerCase()) ||
+          value.toLowerCase().includes(inputValue.toLowerCase())
```

Comment:
any case for fuzzy find here?

product Q, but maybe worth considering since we'll be using this to search lists of titles like all your purchasable content

Reply:
also concerned about matching against the values here. If we're searching tracks, and I input '5', do I really want all tracks with ids containing '5'?

Reply:
great question! i think you are right, the reason for this was that we use the value if there is no label, but to your point, im going to do a label ?? value and then just check for includes on that.